handler/src/main/java/io/netty/handler/ssl/util/OpenJdkSelfSignedCertGenerator.java

```diff
@@ -70,12 +70,12 @@
 
         // Sign the cert to identify the algorithm that's used.
         X509CertImpl cert = new X509CertImpl(info);
-        cert.sign(key, "SHA256withRSA");
+        cert.sign(key, algorithm.equalsIgnoreCase("EC") ? "SHA256withECDSA" : "SHA256withRSA");
```

Comment:
should we verify that only valid `algorithm` names are used ? (Like either `EC` or `RSA`)

Reply:
Added check for EC or RSA. If EC, we'll use `SHA256withECDSA` else `SHA256withRSA` because RSA is last possible case.